packages/imperative/src/events/src/ImperativeEventEmitter.ts

```diff
@@ -0,0 +1,288 @@
+     */
+    private writeEvent(location: string, event: ImperativeEvent) {
+        event.location = location;
+        this.ensureEventsDirExists(location);
+        fs.writeFileSync(join(location, event.type), JSON.stringify(event.toJson(), null, 2));
+    }
+
+    /**
+     * Helper method to create watchers based on event strings and list of callbacks
+     * @param eventName type of event to which we will create a watcher for
+     * @param callbacks list of all callbacks for this watcher
+     * @returns The FSWatcher instance created
+     */
+    private setupWatcher(eventName: string, callbacks: ImperativeEventCallback[] = []): fs.FSWatcher {
+        const dir = this.getEventDir(eventName);
+        this.ensureEventsDirExists(dir);
+        this.ensureEventFileExists(join(dir, eventName));
+
+        const watcher = fs.watch(join(dir, eventName), (event: "rename" | "change") => {
+            // Node.JS triggers this event 3 times
+            const eventContents = this.getEventContents(eventName);
+            const eventTime = eventContents.length === 0 ? "" : (JSON.parse(eventContents) as IImperativeEventJson).time;
+
+            if (this.eventTimes.get(eventName) !== eventTime) {
+                this.logger.debug(`ImperativeEventEmitter: Event "${event}" emitted: ${eventName}`);
+                // Promise.all(callbacks)
+                callbacks.forEach(cb => cb());
+                this.eventTimes.set(eventName, eventTime);
+            }
+        });
+        this.subscriptions.set(eventName, [watcher, callbacks]);
+        return watcher;
+    }
+
+    /**
+     * Check to see if the given event is a User event
+     * @param eventName A string representing the type of event
+     * @returns True if it is a user event, false otherwise
+     */
+    public isUserEvent(eventName: string): eventName is ImperativeEventType {
+        return Object.values<string>(ImperativeUserEvents).includes(eventName);
+    }
+
+    /**
+     * Check to see if the given event is a shared event
+     * @param eventName A string representing the type of event
+     * @returns True if it is a shared event, false otherwise
+     */
+    public isSharedEvent(eventName: string): eventName is ImperativeEventType {
+        return Object.values<string>(ImperativeSharedEvents).includes(eventName);
+    }
+
+    /**
+     * Check to see if the given event is a Custom event
+     * @param eventName A string representing the type of event
+     * @returns True if it is not a zowe or a user event, false otherwise
+     * @internal Not implemented in the MVP
+     */
+    public isCustomEvent(eventName: string): eventName is ImperativeEventType {
+        return !this.isUserEvent(eventName) && !this.isSharedEvent(eventName);
+    }
+
+    /**
+     * ZOWE HOME directory to search for system wide ImperativeEvents like `configChanged`
+     */
+    public getSharedEventDir(): string {
```

Comment:
Could getSharedEventDir and getUserEventDir be private functions? Would the public function getEventDir provide all that is needed for external callers?